can/interfaces/pcan/pcan.py

```diff
@@ -102,7 +108,22 @@ def __init__(self, channel='PCAN_USBBUS1', state=BusState.ACTIVE, bitrate=500000
         else:
             raise ArgumentError("BusState must be Active or Passive")
 
-        result = self.m_objPCANBasic.Initialize(self.m_PcanHandle, pcan_bitrate, hwtype, ioport, interrupt)
+
+        if self.fd:
+            f_clock_val = kwargs.get('f_clock', None)
+            if f_clock_val is None:
+                f_clock = "{}={}".format('f_clock_mhz', kwargs.get('f_clock_mhz', None))
+            else:
+                f_clock = "{}={}".format('f_clock', kwargs.get('f_clock', None))
+            
+            fd_parameters_values = [f_clock] + ["{}={}".format(key, kwargs.get(key, None)) for key in pcan_fd_parameter_list if kwargs.get(key, None) is not None]
+
+            self.fd_bitrate = bytes(' ,'.join(fd_parameters_values), 'ascii')
```

Comment:
Sending and receiving works fine now. Just a small adjustment is needed for python2 compatibility:
> `self.fd_bitrate = bytes(' ,'.join(fd_parameters_values), 'ascii')`
> `TypeError: str() takes at most 1 argument (2 given)`
```suggestion
            self.fd_bitrate = ' ,'.join(fd_parameters_values).encode("ascii")
```